dart/Dockerfile

```diff
@@ -1,15 +1,25 @@
-FROM dart:3.2
+FROM dart:3.2 AS build
 
+# Set the working directory
 WORKDIR /app
 
-ADD pubspec.yaml /app/
-
+# Add pubspec.yaml and get dependencies
+ADD pubspec.* ./
 RUN dart pub get --no-precompile
 
-ADD . /app/
-
+# Copy app source code and AOT compile it
+ADD . .
 RUN dart pub get --offline --no-precompile
 
-WORKDIR /app
+# AOT compile `server.dart` to server executable
+RUN dart compile exe server.dart -o server
+
+# Build minimal serving image from AOT-compiled `server` executable
+FROM scratch
+COPY --from=build /runtime/ /
+COPY --from=build /app/server /app/server
+
+# Fix CI
+COPY --from=build /bin/sh /bin/sh
 
-CMD {{command}}
+CMD {{command}} {{options}}
```

Comment:
what is option `{{options}}` ?

Reply:
This is prepared for other frameworks in the future. Currently shelf and spry do not require it, but there are other frameworks in the dart ecosystem. As far as I know, the `serverpod` framework requires parameters.

Reply:
In addition, {{options}} I copied the Dockerfile from other language frameworks, because the config.yaml configuration document is missing, and I don't know where it comes from.

Reply:
fyi : {{x.y.z}} that you can use in dockerfiles in 
```yaml
x:
  y:
    z:
```
in `config.yaml`